File: Sorting/BubbleSort/Python/bubbleSort.py
def bubbleSort(arr):
    for i in range(len(arr)):
        swapped = False

        for j in range(len(arr)-1-i):
            if(arr[j] > arr[j+1]):
                arr[j], arr[j+1] = arr[j+1], arr[j]
                swapped = True

        if(not swapped):
            break

File: Sorting/BubbleSort/C/bubbleSort.c
void bubbleSort(int* arr){
    int length = sizeof(arr) / sizeof(arr[0]);
    for(int i = 0; i < length; i++){
        bool swapped = false;

        for(int j = 0; j < length-i-1; j++){
            if(arr[j] > arr[j+1]){
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
                swapped = true;
            }
        }

        if(!swapped){
            break;
        }
    }
}
